src/runtime/virtcontainers/fc.go

```diff
@@ -1081,7 +1081,7 @@ func (fc *firecracker) hotplugBlockDevice(ctx context.Context, drive config.Bloc
 
 // hotplugAddDevice supported in Firecracker VMM
 func (fc *firecracker) hotplugAddDevice(ctx context.Context, devInfo interface{}, devType deviceType) (interface{}, error) {
-	span, ctx := fc.trace(ctx, "hotplugAddDevice")
+	span, _ := fc.trace(ctx, "hotplugAddDevice")
```

Comment:
@bergwolf I know this is still a WIP but I'm wondering about changes to what `trace()` is returning. Is there a reason we don't want to return `ctx` here (and in other locations)? This was updated in https://github.com/kata-containers/kata-containers/pull/1389 so that tracing spans would have the appropriate parent/child ordering

Reply:
Because many of these returned `ctx` is not used anymore and causing static checker errors (ineffective assignment). I plan to clean up all these spans properly in future PRs. Some of them are spanning from the passed in `ctx`, some are using a `ctx` saved in either a hypervisor struct, or in a shimv2 service struct. They should follow the same pattern and be unified.

Reply:
@bergwolf In this specific case it is passed to `fc.hotplugBlockDevice(ctx, *devInfo.(*config.BlockDrive), addDevice)` a few lines below it, which is passed to later functions that trace with it. I just created issue https://github.com/kata-containers/kata-containers/issues/1550 to track it.